P4-OVS: Fix issue 82

nupuruttarwar opened the pull request:
On older bash versions == operator doesn't work with [..] operator.
Due to this, depending on bash versions on your system,
install_dep_packages.sh script which install gflags and other
important dependencies will not run and you will run into packages
missing issue

Title: Fix issue 82
Change-type: DefectResolution
Signed-off-by: Nupur Uttarwar <nupur.uttarwar@intel.com>

rst0git commented:
@nupuruttarwar On Fedora 35, I see the following:
```bash
$ get_os_ver_details
$ echo $OS
Fedora Linux
$ [ "$OS" = "Fedora" ] && echo true || echo false
false
$ [[ "$OS" =~ "Fedora" ]] && echo true || echo false
true
```
On Fedora 33:
```bash
$ get_os_ver_details
$ echo $OS
Fedora
```

nupuruttarwar committed:
P4-OVS: Fix issue 82

On older bash versions == operator doesn't work with [..] operator.
Due to this, depending on bash versions on your system,
install_dep_packages.sh script which install gflags and other
important dependencies will not run and you will run into packages
missing issue

Title: Fix issue 82
Change-type: DefectResolution
Signed-off-by: Nupur Uttarwar <nupur.uttarwar@intel.com>

nupuruttarwar commented:
@rst0git Thanks for pointing this out. Since we test on Fedora 33, I missed this. Looking at this https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/5XPTWR5BESKCV52XQ22V3RALTPNO3ENP/, suggestion is to use 'ID' instead of 'NAME' to identify OS from os-release file.
Can you confirm, if /etc/os-release on Fedora 35 had ID=fedora? I will spend some time today to check other OS distributions and versions as well and then push the patch.

rst0git commented:
> Can you confirm, if /etc/os-release on Fedora 35 had ID=fedora?

yes
```bash
$ cat /etc/os-release | grep ID=
ID=fedora
VERSION_ID=35
PLATFORM_ID="platform:f35"
VARIANT_ID=workstation
```

> I will spend some time today to check other OS distributions and versions as well and then push the patch.

You can use a container for testing.

```bash
docker run -it fedora:33 cat /etc/os-release
docker run -it fedora:35 cat /etc/os-release
```
